presto-sqlserver/src/main/java/io/prestosql/plugin/sqlserver/SqlServerClient.java

```diff
@@ -233,6 +240,32 @@ public boolean isLimitGuaranteed(ConnectorSession session)
         return true;
     }
 
+    @Override
+    protected String createTableSql(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
+    {
+        return format("CREATE TABLE %s (%s) WITH (DATA_COMPRESSION = %s)", quoted(remoteTableName), join(", ", columns), getDataCompression(tableMetadata.getProperties()));
+    }
+
+    @Override
+    public Map<String, Object> getTableProperties(JdbcIdentity identity, JdbcTableHandle tableHandle)
+    {
+        ImmutableMap.Builder<String, Object> builder = ImmutableMap.builder();
+
+        try (Connection connection = connectionFactory.openConnection(identity);
+                PreparedStatement statement = connection.prepareStatement("SELECT data_compression_desc FROM sys.partitions p INNER JOIN sys.tables t ON p.object_id = t.object_id WHERE p.partition_number = 1 AND t.name = ?")) {
+            statement.setString(1, tableHandle.getRemoteTableName().getTableName());
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    builder.put(DATA_COMPRESSION, DataCompression.valueOf(resultSet.getString(1)));
```

Comment:
This is not safe. Should SQL Server add new compression mode (say, "BLOCK"), we don't want to fail the method here. I'd rather return no property.

also, is there a test ensuring SHOW CREATE TABLE doesn't return unnecessary data_compression property for all tables now?

Reply:
> is there a test ensuring SHOW CREATE TABLE doesn't return unnecessary data_compression property for all tables now

That property will now be added. I think we should skip it when it is value is unknown (null), when it differs among paritions.